### DISCUSSION TITLE
Add initcap function

### DISCUSSION THREAD

**Discussion by ANelson82** [duckdb/duckdb#12999] [Ideas]
I would love initcap function built in by default.

Teggy posted a macro solution in [discord](https://discord.com/channels/909674491309850675/1032659480539824208/1229488725554102395).

```
CREATE MACRO init_cap(ws) AS
   list_reduce(
      list_transform(
         regexp_split_to_array(ws, '\s+'),
         w -> upper(w[1]) || w[2:]),
      (x,y) -> x || ' ' || y);
```

**Comment by adosib** [comment]
I came up with something similar ([gist](https://gist.github.com/adosib/98873c0c5ffc9a241433c9376c872fc2)), but I think it's more true to Postgres' INITCAP function. It doesn't rely on splitting by whitespace and it down cases letters after the first in the word. 

But it's not super performant as it does more than the snippet macro.

**Comment by johngcrowley** [comment]
In my `select`s, I'm using:
```
substring(<col>, 1, 1) || lower(substring(<col>, 2, strlen(<col>)))
```
and it's handling `null`.

**Reply by bbrewington** [reply]
heads up, this will only handle the first word - other platforms typically perform this operation on each "word" (sequences of alphanumeric characters separated by non-alphanumeric characters...or separated by specified list of delimiters)

**Comment by bbrewington** [comment]
For context on how ubiquitous this function is, here is a list of major data platforms / databases that have it:
- BigQuery
- Databricks
- Snowflake
- Postgres
- Oracle
- Redshift

Cherry-picked the list from dbt's trusted adapters page: https://docs.getdbt.com/docs/trusted-adapters
